packages/core/src/index.js

```diff
@@ -13,16 +13,28 @@ export const merge = (a, b) => {
 
 // sort object-value responsive styles
 const sort = obj => {
-  const next = {}
-  Object.keys(obj)
-    .sort((a, b) => a.localeCompare(b, undefined, {
-      numeric: true,
-      sensitivity: 'base',
-    }))
+  const mediaUnsorted = {}
+  const mediaSorted = {}
+  const rest = {}
+  Object.keys(obj).forEach(key => {
+    if (key.startsWith('@media')) {
+      mediaUnsorted[key] = obj[key]
+    } else {
+      rest[key] = obj[key]
+    }
+  })
+  // Only sort queries that start with @media
+  Object.keys(mediaUnsorted)
```

Comment:
How about removing `forEach` below and directly assigning the sorted value to `mediaSorted`?
```js
const mediaSorted = Object.keys(mediaUnsorted)
  .sort((a, b) =>
    a.localeCompare(b, undefined, {
      numeric: true,
      sensitivity: 'base',
    })
  )
```